### PR TITLE
Update model_generic.mustache, tuple notation breaks when there is only one element in the tuple

### PR DESCRIPTION
**Pull request by edmundloo**
In this excerpt of the mustache template, the matching behaviour is bugged when there is only one element in the tuple notation. A single string with the tuple notation, e.g., `("string")`, will result in a string, and when values are compared with the string, it will do a `contains` match instead of an exact match, which is unintended behaviour.

To reproduce such an issue, create multiple models with oneOf enums that are substrings of each other, like so:
model1 validator:
```
    @field_validator('...')
    def ..._validate_enum(cls, value):
        """Validates the enum"""
        if value not in set(['stringstring']):
            raise ValueError("must be one of enum values ('stringstring')")
        return value
```
model2 validator:
```
    @field_validator('...')
    def ..._validate_enum(cls, value):
        """Validates the enum"""
        if value not in set(['stringstringstring']):
            raise ValueError("must be one of enum values ('stringstring')")
        return value
```
Now if you pass in a response for JSON deserialization, where the value you pass in is "string" or "stringstring", you will match both of these fields and fail the `oneOf` enum check.

We have tested with our codebase, and investigated this due to this error:
```
  File "/Users/edmund/Code/api_resources/models/....py", line 130, in from_json
    raise ValueError("Multiple matches found when deserializing the JSON string into ... with oneOf schemas: ..., ..., .... Details: " + ", ".join(error_messages))
ValueError: Multiple matches found when deserializing the JSON string into ... with oneOf schemas: ..., ..., .... Details: 2 validation errors for ...
...
  Value error, must be one of enum values ('...') [type=value_error, input_value='...', input_type=str]
    For further information visit https://errors.pydantic.dev/2.5/v/value_error
content
  Input should be a valid string [type=string_type, input_value=None, input_type=NoneType]
    For further information visit https://errors.pydantic.dev/2.5/v/string_type
```

We were using some enum lists that only had one value and were breaking when it was only one value. I've modified the template locally and ensured this change is working.

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh ./bin/configs/*.yaml
  ./bin/utils/export_docs_generators.sh
  ``` 
  (For Windows users, please run the script in [Git BASH](https://gitforwindows.org/))
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  IMPORTANT: Do **NOT** purge/delete any folders/files (e.g. tests) when regenerating the samples as manually written tests may be removed.
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (upcoming 7.1.0 minor release - breaking changes with fallbacks), `8.0.x` (breaking changes without fallbacks)
- [x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request. @cbornet (2017/09) @tomplus (2018/10) @krjakbrjak (2023/02) @fa0311 (2023/10) @multani (2023/10)

### CODE REVIEW

**Comment by wing328** [comment]
thanks for the PR.

can you please update the test spec with a test or 2 and regenerate the samples (step 3 in the PR checklist)? 

modules/openapi-generator/src/test/resources/3_0/python/petstore-with-fake-endpoints-models-for-testing.yaml

**Comment by wing328** [comment]
> We were using some enum lists that only had one value and were breaking when it was only one value

Is this supposed to be a constant? do you mind sharing the full spec?

**Comment by edmundloo** [comment]
> > We were using some enum lists that only had one value and were breaking when it was only one value
> 
> Is this supposed to be a constant? do you mind sharing the full spec?

```
    PoopCleaning:
      type: object
      properties:
        task_name:
          type: string
          enum: [cleaning]
        function_name:
          type: string
          enum: [care]
        content:
          type: string
      required:
        - task_name
        - function_name
        - content
    Feeding:
      type: object
      properties:
        task_name:
          type: string
          enum: [cleaning]
        function_name:
          type: string
          enum: [care_nourish]
        content:
          type: string
      required:
        - task_name
        - function_name
        - content
    Bathing:
      type: object
      properties:
        task_name:
          type: string
          enum: [cleaning_deep]
        function_name:
          type: string
          enum: [care_nourish]
        content:
          type: string
      required:
        - task_name
        - function_name
        - content
    Task:
      title: Task
      type: object
      description: Used to test oneOf enums with only one string value.
      properties:
        id:
          type: string
          format: uuid
        activity:
          oneOf:
            - $ref: "#/components/schemas/PoopCleaning"
            - $ref: "#/components/schemas/Feeding"
            - $ref: "#/components/schemas/Bathing"
      required:
        - id
        - activity
```

This part of the spec I just added to the test spec would cause the same issue. In this case with the new changes, the validation functions that support these enums now look like this:
```
    @field_validator('task_name')
    def task_name_validate_enum(cls, value):
        """Validates the enum"""
        if value not in set(['cleaning_deep']):
            raise ValueError("must be one of enum values ('cleaning_deep')")
        return value
```
prior to this change, they looked like this:
```
    @field_validator('task_name')
    def task_name_validate_enum(cls, value):
        """Validates the enum"""
        if value not in ('cleaning_deep'):
            raise ValueError("must be one of enum values ('cleaning_deep')")
        return value
```

In this case, the values defined for `Feeding` and `PoopCleaning` would also pass the validator for `Bathing` since the values of `task_name` for `Feeding` and `PoopCleaning` are both "cleaning" and the statement `if 'cleaning' not in ('cleaning_deep'):` will return False, when it should return True due to the lack of direct match.

**Comment by multani** [comment]
@edmundloo can you re-update the OpenAPI YAML spec file and add only your changes?

At the moment, there's a lot of reformatting in this file, this makes it really hard to see the new spec that is creating issues.

**Comment by edmundloo** [comment]
> @edmundloo can you re-update the OpenAPI YAML spec file and add only your changes?
> 
> At the moment, there's a lot of reformatting in this file, this makes it really hard to see the new spec that is creating issues.

Done!

File: modules/openapi-generator/src/test/resources/3_0/python/petstore-with-fake-endpoints-models-for-testing.yaml [hunk 1]
```diff
@@ -1532,6 +1532,67 @@ components:
...
+      required:
+        - task_name
+        - function_name
+        - content
+    Feeding:
+      type: object
+      properties:
+        task_name:
+          type: string
+          enum: [cleaning]
```
Comment (multani) [inline modules/openapi-generator/src/test/resources/3_0/python/petstore-with-fake-endpoints-models-for-testing.yaml:1555]: Nitpicking, but I think this should be `feeding` instead :) (or something similar)
```suggestion
          enum: [feeding]
```

File: modules/openapi-generator/src/test/resources/3_0/python/petstore-with-fake-endpoints-models-for-testing.yaml [hunk 2]
```diff
@@ -1532,6 +1532,67 @@ components:
...
+        - content
+    Bathing:
+      type: object
+      properties:
+        task_name:
+          type: string
+          enum: [cleaning_deep]
+        function_name:
+          type: string
+          enum: [care_nourish]
```
Comment (multani) [inline modules/openapi-generator/src/test/resources/3_0/python/petstore-with-fake-endpoints-models-for-testing.yaml:1573]: Nitpicking too:
```suggestion
          enum: [care]
```

**Review by multani — CHANGES_REQUESTED** [review]
Can you check the 2 typos in the OpenAPI spec?

Otherwise, I think the change is good and makes sense to me :+1:

**Comment by edmundloo** [comment]
> Can you check the 2 typos in the OpenAPI spec?
> 
> Otherwise, I think the change is good and makes sense to me 👍

Hey @multani , I know it doesn't check out semantically, but these were intentional to ensure sufficient overlap between the different enums to help catch the contains match issues.

**Comment by multani** [comment]
> > Can you check the 2 typos in the OpenAPI spec?
> > Otherwise, I think the change is good and makes sense to me 👍
> 
> Hey @multani , I know it doesn't check out semantically, but these were intentional to ensure sufficient overlap between the different enums to help catch the contains match issues.

Hm OK I see.

@wing328 I think we can merge this fix :+1:

**Review by multani — APPROVED** [review]

**Comment by wing328** [comment]
merged via https://github.com/OpenAPITools/openapi-generator/pull/17749. closing this one

thanks for the contribution.

**Pull request closed without merging**